app/src/main/java/com/metro/gwuexplorer/MainActivity.kt

```diff
@@ -90,54 +69,74 @@ class MainActivity : AppCompatActivity() {
 //                    x++
 //                }
 
-                    val first = results[0]
-                    val firstAdd = first.getAddressLine(0)
-//
-//                //val secondAdd = first.getAddressLine(1)
-//               // Log.d("second Add:", secondAdd)
-//              // val  addr: List<String> = listOf(firstAdd)
-//
-                    addr = mutableListOf(firstAdd)
-
+                first = results[0]
+
+
+            entManager.retrieveNearbyStation(
+                //passing the first result
+                address = first,
+                successCallback = { list,station ->
+                    runOnUiThread {
+                        // Create the adapter and assign it to the RecyclerView
+
+                        val arrayAdapter = ArrayAdapter<String>(this, android.R.layout.select_dialog_singlechoice)
+                        arrayAdapter.addAll(list)
+
+                        if(list.isEmpty()){
+                            Toast.makeText(this@MainActivity, "List is empty...Try again", Toast.LENGTH_LONG).show()
+                        }
+
+                        AlertDialog.Builder(this)
+                            .setTitle("Select an option")
```

Comment:
Make sure all of your strings (for dialogs, toasts, etc.) go into your `strings.xml` file at some point.